File: rustsem-ocaml/cl_tests/tests/t9.m

#allocateInit(2,i32,valcl(1),valcl(2))
#allocateInit(2,i32,valcl(1),valcl(2),valcl(4))
#allocateInit(3,i32,valcl(1),valcl(2),valcl(3))


#allocStack

#allocStack

#allocStack

#write(#sl(0),#own(#blk(0)))

#write(#sl(1),#Shr(#sl(0)))

#read(#sl(1))

#write(#sl(2), #Shr(#sl(0)))

#read(#sl(2))   

#read(#sl(1))  

File: rustsem-ocaml/rust_tests/bindings/t9.rs

struct point{
       x:i32,
       y:i32
}

fn main() {

   let mut p = point {x:5,y:6};

   let b = & mut p;

   (* b).x = 7;

   println!("x:{}, y:{}", (*b).x, (*b).y);

}